### DISCUSSION TITLE
Beta version using Firefox on Mac unsuccessful

### DISCUSSION THREAD

**Discussion by QuinGold** [aclap-dev/video-downloadhelper#48] [Help and Q&A] [answered]
### Your question

I've been using your app for a few years with success. I enroll in a lot of online classes with limited durations, so I download them to study offline. The past year or so I've noticed more videos have video that runs "fast" at over 12,000 fps (I fix them in Premiere). 

So when I saw there was a Beta version of DWH I downloaded it to see if it helped. Unfortunately, I am having no success downloading anything from the training site, not even Audio Only! I updated the Companion App. 

Before I revert back, I'm hoping I'm missing something: 
Mac Sonoma 14.3 using the latest Firefox. Intel chips.

At first it seems to download 100%, but then stops at the Conversion stage. The error is:

Output /Users/--------/VideoDownloads_noBackup/! dwhelper NEW downloads/Name of movie here.mp4 same as Input #0 - exiting
Error opening output file /Users/----------/VideoDownloads_noBackup/! dwhelper NEW downloads/Name of movie here.mp4.
Error opening output files: Invalid argument

My copied settings: 

version: 8.2.0.18a1
channel: beta
build date: 2024-02-01
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: accepted
key (only 16 first characters): 07af69178bff4630
platform: x86-64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:122.0) Gecko/20100101 Firefox/122.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.15",
  "binary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "mac",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/Users/--------",
  "converterBinary": "/Applications/net.downloadhelper.coapp.app/Contents/MacOS/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "n6.1-9-g63b936e978"
}

**Comment by paulrouget** [comment]
Do you mind clicking on the "Details" submenu of the download, and copy paste the values of:

- videoMediaManifest
- audioMediaManifest
- masterManifest

**Reply by QuinGold** [reply]
Thanks for helping. To rule out the training site, I tried downloading from YouTube. Same error. Stops at Converting with a Red "1". When I click on this, I just see "Convert2 error: (Details)" and when I click on Details, I see the error message I pasted above.

How do I click on the Details "submenu" and see this other info? Thanks.

**Reply by QuinGold** [reply]
Oh, I think I found it under Hit Details? I need to download again from the training site.

**Reply by QuinGold** [reply]
Well that's weird. When the YouTube video failed to Convert, I could view "Hit Details" and see those 3 settings (all pointing to google etc). But on the training site, the Hit Details are much shorter and are missing those 3 entries completely.

**Reply by paulrouget** [reply]
> But on the training site, the Hit Details are much shorter and are missing those 3 entries completely.

Do you get a mpd_url field?

**Reply by QuinGold** [reply]
Yes. mpd_url = 
https://71vod-adaptive.akamaized.net/exp=1707336926~acl=%2F17cc64b2-eba5-4ce4-adf2-153515a6dfde%2F%2A~hmac=bb660ead3c5018665c3db9b5651d080a1af0ce368bc9aaa91ef710f704632685/17cc64b2-eba5-4ce4-adf2-153515a6dfde/sep/video/1c2cbcc6,64018190,6d2370b4/audio/43148957,4dcf1e26,d313c2e5/master.mpd?base64_init=1&query_string_ranges=1

**Reply by QuinGold** [reply]
To compare, I downloaded one of the training site videos on my M1 laptop, using the shipping version of the plugin and companion app (not Beta). The video converts, but the Hit Details do not show any field with "manifest".

**Reply by paulrouget** [reply]
Ah, I know what's happening then. It's duplicate of https://github.com/aclap-dev/video-downloadhelper/discussions/42

It's something I'm actively working on. Please subscribe to the above-mentionned issue. I will post there once this is fixed!

Thank you for beta testing :) That's exactly the type of bugs we're hunting for!

**Reply by QuinGold** [reply]
Further info: This training site is streaming videos that are sourced from Vimeo (ie, you upload them to the site and Vimeo "processes" them). The instructor uploads two type of movies: The Zoom replays of our session, and small bonus videos recorded on his iPhone. When I download the Zoom replays, the video plays fast and needs to be fixed. When I download the bonus movies, QuickTime plays them just fine.

**Reply by QuinGold** [reply]
#42. "The answer to Life, the Universe, and Everything".

**Reply by QuinGold** [reply]
More info: 
I download using Download & Convert > Re-encoded MP4 (h264/aac) as that seems to work best on Mac. If I just download directly, the frames would "shake". 
I notice that the movies that failed with an error code, did actually download something. 
However, when I try to open the movies with QuickTime, QT "stops responding" and basically crashes. 
BUT, when I open them in Infuse player or VLC player, they play fine.

**Comment by QuinGold** [comment]
I reverted back to the shipping extension and the older companion app. And restarted the Mac in case QuickTime was just corrupt. Seems okay now.

**Reply by GrampaWildWilly** [reply]
Vimeo has long been a problem for VDH.  I'm glad Paul is looking into it.  (He works for VDH.  I don't.)  The old Google forum is still there.  We're not supposed to post new stuff there but the old stuff still exists & is still relevant & useful.  There's an old thread over there with "too fast" in its title.  If you search over there, you'll find lots of hits on "too fast" because we referred to it so often.  You need to find the thread that is the target of all those references.  We found a way to get the content from Vimeo that VDH can't (yet).  It involves ffmpeg, not the one embedded in the CoApp, ffmpeg from ffmpeg.org.  It is possible to download Vimeo content directly using ffmpeg.  No post-processing is necessary.  One download, video & audio, no conversion needed, plays perfectly.

When you use VDH Download and Convert, you are opening yourself up to something that I say should make you avoid that VDH function.  Unless this has been fixed, Download and Convert can complete a successful download, then fail during the conversion.  That causes VDH to delete the perfectly good file it downloaded.  It might be advisable to just do a regular Download.  After you have that completed successfully, do your Convert.  If the Convert fails for whatever reason, you still have the downloaded file & you can try a different set of conversion parameters to see if you can make the conversion succeed.  You won't have to download the file again, which would just be a waste of time & bandwidth.

**Reply by QuinGold** [reply]
Thanks, I remember reading those threads on too fast. I wasn't able to figure out the solutions offered, but being familiar with video editing, I just set the frame rate for the source to 25 fps, add the audio, then render again in Premiere. Fortunately there is usually only a handful of videos for each class I take, and some of them work fine (for instance, only 1 video was fast in the recent class). What confuses me is why some are fine and some are fast. I sometimes restart the Mac just in case it might help, but it doesn't.

**Comment by QuinGold** [comment, marked as answer]
YAY! I just installed the latest Beta (Feb 8) and now I can just click Quick Download instead of Download & Convert, and the movie does NOT shudder! This has never happened before - I always have to "Convert". And the movies that were running FAST are not playing normally! 
I did try a Download & Convert as I would normally, and it still gave a red error as soon as it switched over to the Convert step. However, this time the saved movie did not crash Quicktime when I opened it, and it seemed fine.

**Comment by QuinGold** [comment]
I emailed you a link to try. 
I also retested everything I said above with two Macs running the older and beta versions to compare, and the only bug now is that the Convert option stops with a red error (the movie seems fine though). It's not a big deal to me as now I don't have to Convert on Mac! Thank you thank you thank you!

**Comment by QuinGold** [comment]
Another small bug: When downloading a video that is 1:36:11 long, the duration omits the "hour" and just shows 36:11 duration.
